backend/handlers/post_collection_handler.py

```diff
@@ -72,26 +72,38 @@ def create_post(post_data, user, institution):
 
         entity_type = PostFactory.get_type(post_data)
 
-        for follower in institution.followers:
-            if follower != user.key:
-                send_message_notification(
-                    follower.urlsafe(),
-                    user.key.urlsafe(),
-                    entity_type,
-                    post.key.urlsafe(),
-                    user.current_institution
-                )
+        params = {
+            'sender_key': user.key.urlsafe(),
+            'entity_key': post.key.urlsafe(),
+            'entity_type': entity_type,
+            'institution_key': post.institution.urlsafe(),
+            'current_institution': user.current_institution.urlsafe()
+        }
+
+        enqueue_task('notify-followers', params)
 
         if(post.shared_post):
+            shared_post = post.shared_post.get()
             entity_type = 'SHARED_POST'
             params = {
-                'receiver_key': post.author.urlsafe(),
+                'receiver_key': shared_post.author.urlsafe(),
                 'sender_key': user.key.urlsafe(),
                 'entity_key': post.key.urlsafe(),
                 'entity_type': entity_type,
-                'current_institution': user.current_institution.urlsafe()
+                'current_institution': user.current_institution.urlsafe(),
+                'subscribers': json.dumps([subscriber.urlsafe() for subscriber in shared_post.subscribers])
```

Comment:
This "for" should not be done in the handler, subscribers list could lead to the whole number of users of the system and can take too much time to compute.